ibm_mq/datadog_checks/ibm_mq/ibm_mq.py

```diff
@@ -0,0 +1,113 @@
+# (C) Datadog, Inc. 2018
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+
+from datadog_checks.checks import AgentCheck
+
+from six import iteritems
+
+try:
+    import pymqi
+except ImportError:
+    pymqi = None
+
+from . import errors, metrics
+
+
+class IbmMqCheck(AgentCheck):
+
+    METRIC_PREFIX = 'ibm_mq'
+
+    SERVICE_CHECK = 'ibm_mq.can_connect'
+
+    QUEUE_MANAGER_SERVICE_CHECK = 'ibm_mq.queue_manager'
+    QUEUE_SERVICE_CHECK = 'ibm_mq.queue.can_connect'
+
+    def check(self, instance):
+        if not pymqi:
+            self.log.error("You need to install pymqi")
+            raise errors.PymqiException("You need to install pymqi")
+
+        channel = instance.get('channel')
+        queue_manager_name = instance.get('queue_manager', 'default')
+
+        host = instance.get('host')
+        port = instance.get('port')
+
+        host_and_port = "{}({})".format(host, port)
+
+        username = instance.get('username')
+        password = instance.get('password')
+
+        custom_tags = instance.get('tags', [])
+
+        tags = [
+            "queue_manager:{}".format(queue_manager_name),
+            "host:{}".format(host),
+            "port:{}".format(port),
+            "channel:{}".format(channel)
+        ]
+
+        tags += custom_tags
```

Comment:
nit: `tags.extend(custom_tags)`

Reply:
I prefer this syntax

Reply:
it's very common in our codebase, I'm gonna leave it

Reply:
Up to you! Though we definitely use the actual method more:

```console
$ ddev run rg --no-filename -c "tag[a-zA-Z_]*\.extend" | python -c "import sys;print(sum(int(line.strip()) for line in sys.stdin))"
70
$ ddev run rg --no-filename -c "tag[a-zA-Z_]* ?\+= ?[a-zA-Z_]" | python -c "import sys;print(sum(int(line.strip()) for line in sys.stdin))"
36
```